kornia/enhance/jpeg.py

```diff
@@ -0,0 +1,566 @@
+    """
+    # Convert RGB image to YCbCr.
+    image_ycbcr: Tensor = rgb_to_ycbcr(image_rgb)
+    # Scale pixel-range to [0, 255]
+    image_ycbcr = 255.0 * image_ycbcr
+    # Perform chroma subsampling
+    input_y, input_cb, input_cr = _chroma_subsampling(image_ycbcr)
+    # Patchify, DCT, and rounding
+    input_y, input_cb, input_cr = _patchify_8x8(input_y), _patchify_8x8(input_cb), _patchify_8x8(input_cr)
+    dct_y, dct_cb, dct_cr = _dct_8x8(input_y), _dct_8x8(input_cb), _dct_8x8(input_cr)
+    y_encoded: Tensor = _quantize(
+        dct_y,
+        jpeg_quality,
+        quantization_table_y,
+    )
+    cb_encoded: Tensor = _quantize(
+        dct_cb,
+        jpeg_quality,
+        quantization_table_c,
+    )
+    cr_encoded: Tensor = _quantize(
+        dct_cr,
+        jpeg_quality,
+        quantization_table_c,
+    )
+    return y_encoded, cb_encoded, cr_encoded
+
+
+def _jpeg_decode(
+    input_y: Tensor,
+    input_cb: Tensor,
+    input_cr: Tensor,
+    jpeg_quality: Tensor,
+    H: int,
+    W: int,
+    quantization_table_y: Tensor,
+    quantization_table_c: Tensor,
+) -> Tensor:
+    """Performs JPEG decoding.
+
+    Args:
+        input_y (Tensor): Compressed Y component of the shape :math:`(B, N, 8, 8)`.
+        input_cb (Tensor): Compressed Cb component of the shape :math:`(B, N, 8, 8)`.
+        input_cr (Tensor): Compressed Cr component of the shape :math:`(B, N, 8, 8)`.
+        jpeg_quality (Tensor): Compression strength of the shape :math:`(B)`.
+        H (int): Original image height.
+        W (int): Original image width.
+        quantization_table_y (Tensor): Quantization table for Y channel.
+        quantization_table_c (Tensor): Quantization table for C channels.
+
+    Returns:
+        rgb_decoded (Tensor): Decompressed RGB image of the shape :math:`(B, 3, H, W)`.
+    """
+    # Dequantize inputs
+    input_y = _dequantize(
+        input_y,
+        jpeg_quality,
+        quantization_table_y,
+    )
+    input_cb = _dequantize(
+        input_cb,
+        jpeg_quality,
+        quantization_table_c,
+    )
+    input_cr = _dequantize(
+        input_cr,
+        jpeg_quality,
+        quantization_table_c,
+    )
+    # Perform inverse DCT
+    idct_y: Tensor = _idct_8x8(input_y)
+    idct_cb: Tensor = _idct_8x8(input_cb)
+    idct_cr: Tensor = _idct_8x8(input_cr)
+    # Reverse patching
+    image_y: Tensor = _unpatchify_8x8(idct_y, H, W)
+    image_cb: Tensor = _unpatchify_8x8(idct_cb, H // 2, W // 2)
+    image_cr: Tensor = _unpatchify_8x8(idct_cr, H // 2, W // 2)
+    # Perform chroma upsampling
+    image_cb = _chroma_upsampling(image_cb)
+    image_cr = _chroma_upsampling(image_cr)
+    # Back to [0, 1] pixel-range
+    image_ycbcr: Tensor = torch.stack((image_y, image_cb, image_cr), dim=1) / 255.0
+    # Convert back to RGB space.
+    rgb_decoded: Tensor = ycbcr_to_rgb(image_ycbcr)
+    return rgb_decoded
+
+
+def diff_jpeg(
```

Comment:
as the high level api for this module, i would like here to have a more explicit name like `encode_to_jpeg_differentiable`

Reply:
Agree, but `encode_to_jpeg_differentiable` is somewhat misleading since we perform both encoding and decoding. What about `jpeg_encode_decode_differentiable`?

Reply:
I change the naming to `jpeg_encode_decode_differentiable` and `JPEGEncodeDecodeDifferentiable` for the module. Let me know what you think :)

Reply:
got it -- in that case to have a more concise name `jpeg_codec_differentiable` where codec usually referred to compress/decompress

Reply:
Done :)